### PR TITLE
Fix: Force stereo when opening audio device

### PR DESCRIPTION
**Pull request by SzQ1**
#### Summary
Bugfixes "Fixes Phantom metallic sound"

#### Purpose of change
Fixes "metallic" pitch-shifted sounds (e.g. walking) on some sound cards.

#### Describe the solution

I had to deep dive SDL2 again, and in the process noticed that Mix_OpenAudio() by default allows SDL2 to choose any number of sound channels. For some sound devices, this number may be higher than 2 (e.g. 4 or 6).

When sound is to be played, it is first loaded via SDL2, which converts it into the mixer's format. If played sound must be pitch-shifted, it is then passed to Cata's do_pitch_shift() which assumes there are always 2 channels, and in case there are not 2 channels - scrambles the sound and turns it "metallic".

You can reproduce "metallic" effect on system with any sound card by changing int audio_channels to 6 and then launching the game with SDL_AUDIODRIVER=disk environment variable. SDL2 will create a raw file in game's directory and use it as output device that matches desired specs, and afterwards the file can be imported into Audacity (16-bit PCM, LE, 6 channels, 44100) and replayed to get exactly this:
https://user-images.githubusercontent.com/74045419/112876475-48654500-9111-11eb-8fd4-b57b76d8c415.mp4

Solution is to use Mix_OpenAudioDevice(), which allows specifying what specs the device must match, and force device to be stereo.

#### Describe alternatives you've considered
Fixing do_pitch_shift() for non-stereo, but I doubt surround sound is something a roguelike needs to support. Also, less channels == less RAM consumed by each sound sample.

#### Testing
I don't have the hardware myself, but @EirenexTheDragon confirmed this patch fixes sounds.
#### Additional context
Code stolen from @olanti-p. Then transformed also by Olanti to work with DDA. 
Title and description of PR stolen from Olanti as well, from here:
https://github.com/cataclysmbnteam/Cataclysm-BN/pull/863

### CODE REVIEW

**Comment by Maleclypse** [comment]
I put your summary into the format if you need to change any of the words inside the " " it's fine.

**Comment by Maleclypse** [comment]
Just noticed you need to astyle your code.  Once you do so, please ping me and I'll kick your tests to start running again.  It looks like they'll pass fine, but the repo settings require that all tests listed as required have passed on the final commit to merge.

File: src/sdlsound.cpp [hunk 1]
```diff
@@ -101,7 +101,7 @@ bool init_sound()
     // We should only need to init once
     if( !sound_init_success ) {
         // Mix_OpenAudio returns non-zero if something went wrong trying to open the device
-        if( !Mix_OpenAudio( audio_rate, audio_format, audio_channels, audio_buffers ) ) {
+        if( !Mix_OpenAudioDevice( audio_rate, audio_format, audio_channels, audio_buffers, nullptr, SDL_AUDIO_ALLOW_FREQUENCY_CHANGE ) ) {
```
Comment (kevingranade) [inline src/sdlsound.cpp:104]:
```suggestion
        if( !Mix_OpenAudioDevice( audio_rate, audio_format, audio_channels, audio_buffers, nullptr,
            SDL_AUDIO_ALLOW_FREQUENCY_CHANGE ) ) {
```

File: src/sdlsound.cpp [hunk 2]
```diff
@@ -101,7 +101,8 @@ bool init_sound()
     // We should only need to init once
     if( !sound_init_success ) {
         // Mix_OpenAudio returns non-zero if something went wrong trying to open the device
-        if( !Mix_OpenAudio( audio_rate, audio_format, audio_channels, audio_buffers ) ) {
+        if( !Mix_OpenAudioDevice( audio_rate, audio_format, audio_channels, audio_buffers, nullptr,
+            SDL_AUDIO_ALLOW_FREQUENCY_CHANGE ) ) {
```
Comment (kevingranade) [inline src/sdlsound.cpp:105]:
```suggestion
        if( !Mix_OpenAudioDevice( audio_rate, audio_format, audio_channels, audio_buffers, nullptr,
                                  SDL_AUDIO_ALLOW_FREQUENCY_CHANGE ) ) {
```

Comment (SzQ1) [reply src/sdlsound.cpp]: Thanks for help, I'm still getting lost on github.

**Pull request merged**